### PR TITLE
Mehrzad_66512_update balance tradeEngine

### PR DESCRIPTION
**Pull request by mehrzad-fs**

### CODE REVIEW

File: src/botPage/bot/TradeEngine/Balance.js [hunk 1]
```diff
@@ -1,28 +1,26 @@
...
+  api.onMessage().subscribe(({ data }) => {
+    if (data?.error?.code) {
+      return;
     }
-    // eslint-disable-next-line class-methods-use-this
-    getBalance(type) {
-      const balance = globalObserver.getState("balance");
-      const balance_str = `${balance}`;
-      return type === "STR" ? balance_str : Number(balance);
+    if (data?.msg_type === "balance") {
```
Comment (maryiafrantsava-binary) [inline src/botPage/bot/TradeEngine/Balance.js:17]:
```suggestion
    if (data && data?.msg_type === "balance") {
```

Comment (mehrzad-fs) [reply src/botPage/bot/TradeEngine/Balance.js:16]: > 

Optional chaining is for reading the value of a property located deep within a chain of connected objects without having to check that each reference in the chain is valid
So we don't need to double-check data here
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining

Comment (maryiafrantsava-binary) [reply src/botPage/bot/TradeEngine/Balance.js:16]: @mehrzad-fs but track js, in this case, can catch the error when the variable data will be undefined inside of the if block. We are faced with like these issues very often...

Comment (maryiafrantsava-binary) [reply src/botPage/bot/TradeEngine/Balance.js:16]: maybe any other case, but i remember something like this

**Review by sandeep-deriv — APPROVED** [review]
LGTM 👌

**Pull request closed without merging**